avocado/core/job.py

```diff
@@ -581,6 +581,19 @@ def cleanup(self):
         self.__stop_job_logging()
         if self.__remove_tmpdir and os.path.exists(self.tmpdir):
             shutil.rmtree(self.tmpdir)
+        if (getattr(self.args, "dry_run", False) and not
+                getattr(self.args, "dry_run_no_cleanup", False)):
+            # Also clean up temp base directory created because of the dry-run
+            base_logdir = getattr(self.args, "base_logdir", None)
+            if base_logdir is not None:
+                try:
```

Comment:
I know `try .. except` is supposed to be used as a common flow controller in Python. But in this case wouldn't be better (less code, more readable) use `os.path.exists`?

Reply:
The reason for `try .. except` is because things can happen between a `if os.path.exists()` and the actual removal... filesystems are shared resources and we can't assume exclusive control.  Other process may be doing exactly the same thing.  If what we need is the directory deleted (which is true in this case), then a `try/except` block, is safer.